.checkstyle/checkstyle.xml

```diff
@@ -0,0 +1,140 @@
+        <module name="ConstantName">
+            <property name="format" value="(^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$)"/>
+        </module>
+        <module name="LocalVariableName"/>
+        <module name="LocalFinalVariableName"/>
+        <module name="MemberName"/>
+        <module name="ClassTypeParameterName">
+            <property name="format" value="^[A-Z][a-zA-Z0-9]*$$"/>
+        </module>
+        <module name="MethodTypeParameterName">
+            <property name="format" value="^[A-Z][a-zA-Z0-9]*$$"/>
+        </module>
+        <module name="InterfaceTypeParameterName">
+            <property name="format" value="^[A-Z][a-zA-Z0-9]*$$"/>
+        </module>
+        <module name="PackageName"/>
+        <module name="ParameterName"/>
+        <module name="StaticVariableName"/>
+        <module name="TypeName"/>
+        <module name="AvoidStarImport"/>
+
+        <!-- dependencies -->
+        <!--<module name="ImportControl">
+            <property name="file" value="${importControlFile}"/>
+        </module>-->
+
+        <!-- whitespace -->
+        <module name="GenericWhitespace"/>
+        <module name="NoWhitespaceBefore"/>
+        <module name="WhitespaceAfter" />
+        <module name="NoWhitespaceAfter"/>
+        <module name="WhitespaceAround">
+            <property name="allowEmptyConstructors" value="true"/>
+            <property name="allowEmptyMethods" value="true"/>
+        </module>
+        <module name="Indentation"/>
+        <module name="MethodParamPad"/>
+        <module name="ParenPad"/>
+        <module name="TypecastParenPad"/>
+
+        <!-- locale-sensitive methods should specify locale -->
+        <module name="Regexp">
+            <property name="format" value="\.to(Lower|Upper)Case\(\)"/>
+            <property name="illegalPattern" value="true"/>
+            <property name="ignoreComments" value="true"/>
+        </module>
+
+        <!-- code quality -->
+        <module name="MethodLength"/>
+        <module name="ParameterNumber">
+            <!-- default is 8 -->
+            <property name="max" value="13"/>
+        </module>
+        <module name="ClassDataAbstractionCoupling">
+            <!-- default is 7 -->
+            <property name="max" value="20"/>
+        </module>
+        <module name="BooleanExpressionComplexity">
+            <!-- default is 3 -->
+            <property name="max" value="5"/>
+        </module>
+
+        <module name="ClassFanOutComplexity">
+            <!-- default is 20 -->
+            <property name="max" value="44"/>
+        </module>
+        <module name="CyclomaticComplexity">
+            <!-- default is 10-->
+            <property name="max" value="19"/>
+        </module>
+        <module name="JavaNCSS">
+            <!-- default is 50 -->
+            <property name="methodMaximum" value="100"/>
+        </module>
+        <module name="NPathComplexity">
+            <!-- default is 200 -->
+            <property name="max" value="5832"/>
```

Comment:
Any reason why not use the defaults?

Reply:
I copied these from the Strimzi kafka operator repo, happy to leave it as the defaults instead since this project will be a lots smaller and simpler

Reply:
Yeah, on the beginning of Strimzi we were changing these values. But it makes little sense. We later turned to use annotations to suppress the issues where it made sense rather than changing the defaults. So I think maybe starting with the originaldefaults might make more sense.

Reply:
@scholzj I've removed the overrides